src/app/pages/components/tree/tree.component.ts

```diff
@@ -29,5 +32,64 @@ export class TreeComponent {
       }],
     }],
   }];
-
+  options: ITreeOptions = {
+    displayField: 'name',
+    isExpandedField: 'expanded',
+    idField: 'id',
+    hasChildrenField: 'children',
+    actionMapping: {
+      mouse: {
+        dblClick: (tree, node, $event) => {
+          if (node.hasChildren) TREE_ACTIONS.TOGGLE_EXPANDED(tree, node, $event);
+        },
+      },
+    },
+    animateExpand: true,
+    animateSpeed: 30,
+    animateAcceleration: 1.2,
+  }
+  constructor(
+    private modalService: NgbModal,
+  ) { }
+  ngOnInit() {
+  }
+  @ViewChild(TreeComponent)
+  private tree: TreeComponent;
+  actionClick(node, tree, op) {
+    switch (op) {
+      case 1:
+        this.openPopup(node, tree , 'Edit' , 1);
+        break;
+      case 2:
+        this.openPopup(node, tree , 'Add Node' , 2);
+        break;
+      case 3:
+        this.openPopup(node, tree , 'Add Catagery', 3);
+        break;
+      default: alert('Invalid Action');
+    }
+  }
+  openPopup(node , tree , header , ops) {
+    const activeModal = this.modalService.open(InputComponent, { size: 'lg', container: 'nb-layout' });
+    activeModal.componentInstance.modalHeader = header;
+    activeModal.componentInstance.modalOperation = ops;
+    activeModal.componentInstance.modalNode = node;
+    activeModal.componentInstance.modalTree = tree;
+    activeModal.result.then(() => {
+      this.tree.treeModel.update();
+    }, () => {
+    });
+  }
+  deleteNode(node, tree) {
+    if (confirm('Are you sure to delete ' + node.data.name)) {
+      const parentNode = node.realParent ? node.realParent : node.treeModel.virtualRoot;
+      _.remove(parentNode.data.children, function (child) {
```

Comment:
there is no need to use lodash here, simple `filter` will work just fine

Reply:
ok dear i will change it

Reply:
did know how to do that

Reply:
Something like:

```typescript
parentNode.data.children = parentNode.data.children.filter((child) => child !== node.data);
```

Reply:
don't know how to do that

Reply:
```typescript
parentNode.data.children = parentNode.data.children.filter((child) => child !== node.data);
```